File: algorithms/0551.student-attendance-record-i/solution.go
func checkRecord(s string) bool {
	a := 0
	l := 0

	for _, r := range s {
		switch r {
		case 'A':
			a++
			l = 0
		case 'L':
			l++
		default:
			l = 0
		}

		if a >= 2 || l >= 3 {
			return false
		}
	}

	return true
}

File: algorithms/0551.student-attendance-record-i/solution.rs
impl Solution {
    pub fn check_record(s: String) -> bool {
        let mut a = 0;
        let mut l = 0;

        for c in s.chars() {
            match c {
                'A' => {
                    a += 1;
                    l = 0;
                }
                'L' => l += 1,
                _ => l = 0,
            }

            if a >= 2 || l >= 3 {
                return false;
            }
        }

        true
    }
}
